### PR TITLE
Precompiled headers

### PR DESCRIPTION
**Pull request by Holzhaus**
*Based on #12025.*

No profiling done, I just selected a bunch of headers from the `util` subdirectory plus some other ones that I though might be widely used and don't change that often.

Ideally we'd also add this:

```cmake
target_precompile_headers(mixxx-test REUSE_FROM mixxx-lib)
```

However, this does not work because I think the `libshout-idjc` build messes with the compile options, so the precompiled headers are not reusable and need to be recompiled.

### CODE REVIEW

**Comment by daschuer** [comment]
> I think the libshout-idjc build messes with the compile options

Would it be than reasonable to move libshout-idjc to a separate library?

**Comment by daschuer** [comment]
Not sure why the https://github.com/mixxxdj/mixxx/pull/12025 commits are still listed.
Can you rebase it to get rid of them?

File: CMakeLists.txt [hunk 1]
```diff
@@ -1093,7 +1093,6 @@ add_library(mixxx-lib STATIC EXCLUDE_FROM_ALL
   src/util/logger.cpp
   src/util/logging.cpp
   src/util/mac.cpp
-  src/util/moc_included_test.cpp
```
Comment (daschuer) [inline CMakeLists.txt:1096]: Why does this no longer work? Does it error with a false positive?

Comment (Holzhaus) [reply CMakeLists.txt:1096]: It always fails on my machine. If you want to try yourself, just checkout this branch and remove the commit to see if it works for you.

**Comment by Holzhaus** [comment]
> Not sure why the https://github.com/mixxxdj/mixxx/pull/12025 commits are still listed.
> Can you rebase it to get rid of them?  

Done.

**Comment by JoergAtGithub** [comment]
The Windows build failure has it's origin in this name collision:
grafik

Renaming the variable interface fixed the issue. The build of Mixxx takes 1:50 instead of 6 minutes now. And more than 30 seconds of this time is for libdjinterop.

**Comment by JoergAtGithub** [comment]
Tested various build targets locally all work flawless with a massive improvement in build time.
Full build of x64_off is down to 1'20
Full build of x64_legacy is down to 1'55
Code LGTM and CI passes on all platforms. Thank you!

**Pull request merged**